salt/roster/cache.py

```diff
@@ -18,38 +51,53 @@
       - ipv4-local    # loopback addresses
 
 
-Various other keys are also supported:
+This is the default ``roster_order``.
+It prefers IPv6 over IPv4 addresses and private addresses over public ones.
+The relevant data will be fetched from the cache in-order, and the first match will fill the ``host`` key.
+
+Other address selection parameters are also possible:
+
 .. code-block:: yaml
 
-# Examples
 roster_order:
   host:
-    - id                             # use the minion id
     - global|public|private|local    # Both IPv6 and IPv4 addresses in that range
-    - grain: fqdn_ip4                # lookup this grain
-    - pillar:                        # Works with pillardata too
-        - dns:fqdn                   # ':' syntax and lists supported
-    - mine: network.ip_addrs         # Mine data lookup works the same
-    - 2000::/3                       # CIDR networks, both IPv4 and IPv6 supported
+    - 2000::/3                       # CIDR networks, both IPv4 and IPv6 are supported
 
-                                     # The lookup can be used for arbitrary other keys as well,
-  user: pillar: ssh:auth:user
-  priv:
-    - pillar: ssh:auth:private_key
 
+Using cached data
+=================
 
-.. versionadded:: Nitrogen
-    expanded functionality to include arbitrary pillar and grain keys, IPv6 addresses
-    allow roster_order per key in config
-    changed default order in line with os defaults
+Several cached libraries can be selected using the ``library: `` prefix, followed by the library key.
+This can be referenced using the same ``:`` syntax as e.g. ``pillar.get()``.
```

Comment:
You can actually make this a direct link in the docs by replacing ``pillar.get()`` with ``:py:func:`pillar.get <salt.modules.pillar.get>` ``. So this line would change to:

```
This can be referenced using the same ``:`` syntax as e.g. :py:func:`pillar.get <salt.modules.pillar.get>`.
```